iocore/net/SNIActionPerformer.cc

```diff
@@ -0,0 +1,96 @@
+/** @file
+
+  Implementation of SNIActionPerformer
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include "P_SNIActionPerformer.h"
+#include "tscore/ts_file.h"
+#include "tscpp/util/TextView.h"
+#include "tscore/BufferWriter.h"
+#include "tscore/bwf_std_format.h"
+
+SNI_IpAllow::SNI_IpAllow(std::string &ip_allow_list, std::string const &servername)
+{
+  ts::TextView content{ip_allow_list};
+  if (content && content[0] == '@') {
+    std::error_code ec;
+    ts::file::path path{content.remove_prefix(1)};
+    ts::LocalBufferWriter<1024> w;
```

Comment:
This buffer is only needed when an error occurs.

Reply:
delayed the construction of `w` to the error path.